File: cs-budget-api/main/src/Entities/Transaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Entities;

public class Transaction
{
    public Guid Id { get; init; }

    [MaxLength(50)]
    public required string Category { get; set; }

    [MaxLength(200)]
    public required string Description { get; set; }

    public decimal Value { get; set; }

    public DateTime Timestamp { get; set; }

    [ForeignKey(nameof(CredentialId))]
    public virtual Credential Credential { get; set; } = null!;

    public required Guid CredentialId { get; set; }
}

File: go-budget-api/internal/entities/transaction.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Transaction struct {
	Id           string     `gorm:"type:uuid;primary_key"`
	Category     string     `gorm:"type:varchar(50);NOT NULL"`
	Description  string     `gorm:"type:varchar(200);NOT NULL"`
	Value        float64    `gorm:"type:decimal;NOT NULL"`
	Timestamp    time.Time  `gorm:"type:timestamp(3);NOT NULL"`
	Credential   Credential `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
	CredentialId string     `gorm:"NOT NULL"`
}

func (t *Transaction) BeforeCreate(tx *gorm.DB) (err error) {
	t.Id = uuid.New().String()
	return
}
